### PR TITLE
Add phrase keys

### PR DESCRIPTION
**Pull request by sebastijansakac**

### CODE REVIEW

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/OrderManager.php  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 1]
```diff
@@ -71,21 +84,17 @@ public function createOrder($cart, $state, $paymentType)
...
             $orderState = new \OrderState();
             $orderState->name = array();
-            foreach (\Language::getLanguages() as $language) {
-                if (\Tools::strtolower($language['iso_code']) == 'de') {
-                    $orderState->name[$language['id_lang']] = $orderStateInfo['de'];
-                } else {
-                    $orderState->name[$language['id_lang']] = $orderStateInfo['en'];
-                }
+            foreach (\Language::getLanguages(false) as $language) {
+                $orderStateInfo = $this->module->getTranslationForLanguage($language['iso_code'], $key, self::TRANSLATION_FILE);
```
Comment (rinnhofer) [inline wirecardpaymentgateway/helper/OrderManager.php]: Codacy Issue found: [Line exceeds 120 characters; contains 128 characters](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 2]
```diff
@@ -77,15 +90,12 @@ public function createOrder($cart, $state, $paymentType)
...
             $orderState = new \OrderState();
             $orderState->name = array();
-            foreach (\Language::getLanguages() as $language) {
-                if (\Tools::strtolower($language['iso_code']) == 'de') {
-                    $orderState->name[$language['id_lang']] = $orderStateInfo['de'];
-                } else {
-                    $orderState->name[$language['id_lang']] = $orderStateInfo['en'];
-                }
+            foreach (\Language::getLanguages(false) as $language) {
+                $orderStateInfo = $this->module->getTranslationForLanguage($language['iso_code'], $stateKey, self::TRANSLATION_FILE);
```
Comment (rinnhofer) [inline wirecardpaymentgateway/helper/OrderManager.php]: Codacy Issue found: [Line exceeds 120 characters; contains 133 characters](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066)

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 3]
```diff
@@ -77,15 +90,12 @@ public function createOrder($cart, $state, $paymentType)
     public function createOrderState($stateName)
```
Comment (ozada) [inline wirecardpaymentgateway/helper/OrderManager.php]: stateName could be state or stateKey

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 4]
```diff
@@ -77,15 +90,12 @@ public function createOrder($cart, $state, $paymentType)
...
     {
         if (!\Configuration::get($stateName)) {
-            $orderStateInfo = $this->getOrderStateInfo($stateName);
+            $stateKey = $this->getStateKeyFromOrderState($stateName);
```
Comment (ozada) [inline wirecardpaymentgateway/helper/OrderManager.php]:
```suggestion
            $translationKey = $this->getStateKeyFromOrderState($stateKey);
```

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 5]
```diff
@@ -97,38 +107,28 @@ public function createOrderState($stateName)
             $orderState->add();
 
             \Configuration::updateValue(
-                $stateName,
+                $stateKey,
```
Comment (ozada) [inline wirecardpaymentgateway/helper/OrderManager.php]: why you changed it to **stateKey**. It update/create WIRECARD_OS_KEY_* or phraseapp key in DB?

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 6]
```diff
@@ -23,10 +23,23 @@
...
+     * @var string
+     * @since 2.8.0
+     */
+    const TRANSLATION_FILE = "ordermanager";
+
     const WIRECARD_OS_STARTING = 'WIRECARD_OS_STARTING';
     const WIRECARD_OS_AWAITING = 'WIRECARD_OS_AWAITING';
     const WIRECARD_OS_AUTHORIZATION = 'WIRECARD_OS_AUTHORIZATION';
 
+    const WIRECARD_OS_KEY_STARTED = 'order_state_payment_started';
```
Comment (ozada) [inline wirecardpaymentgateway/helper/OrderManager.php]:
```suggestion
    const WIRECARD_OS_TRANSLATION_KEY_STARTED = 'order_state_payment_started';
```
I think its make sense define it so, cause **WIRECARD_OS_KEY_*** could be interpreted as key of object.

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 7]
```diff
@@ -97,38 +107,28 @@ public function createOrderState($stateName)
...
     }
 
     /**
-     * Getter for language texts to specific order state
-     *
      * @param $stateName
-     * @return array
-     * @since 1.0.0
+     * @return string
+     * @throws Exception
```
Comment (ozada) [inline wirecardpaymentgateway/helper/OrderManager.php]: since tag is missing.

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 8]
```diff
@@ -97,38 +107,28 @@ public function createOrderState($stateName)
...
+                $stateKey,
                 (int)($orderState->id)
             );
         }
     }
 
     /**
-     * Getter for language texts to specific order state
-     *
      * @param $stateName
```
Comment (ozada) [inline wirecardpaymentgateway/helper/OrderManager.php]: type hinting is missing

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 9]
```diff
@@ -97,39 +117,27 @@ public function createOrderState($stateName)
...
-                    'en' => 'Wirecard payment authorized',
-                );
-            case self::WIRECARD_OS_AWAITING:
-            default:
-                return array(
-                    'de' => 'Wirecard Bezahlung ausständig',
-                    'en' => 'Wirecard payment awaiting'
-                );
+        $translationKeys = self::ORDER_STATE_TRANSLATION_KEY_MAP;
+        if (!isset($keys[$state])) {
```
Comment (rinnhofer) [inline wirecardpaymentgateway/helper/OrderManager.php]: Codacy Issue found: [Avoid unused local variables such as '$keys'.](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Coverage increased per file
===========================
+ wirecardpaymentgateway/helper/OrderManager.php  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Coverage increased per file
===========================
+ wirecardpaymentgateway/helper/OrderManager.php  3
+ wirecardpaymentgateway/models/PaymentGuaranteedInvoiceRatepay.php  3
+ wirecardpaymentgateway/classes/Service/TabManagerService.php  3
+ wirecardpaymentgateway/helper/Service/ContextService.php  1
+ wirecardpaymentgateway/classes/Response/Cancel.php  4
+ wirecardpaymentgateway/helper/AdditionalInformationBuilder.php  4
+ wirecardpaymentgateway/classes/Response/Initial/Success.php  3
+ wirecardpaymentgateway/classes/Notification/Success.php  4
+ wirecardpaymentgateway/classes/Config/CreditcardConfigurationFactory.php  9
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/models/CreditCardVault.php  2
+ wirecardpaymentgateway/models/Payment.php  1
+ wirecardpaymentgateway/classes/Transaction/Builder/Entity/EntityBuilderFactory.php  19
+ wirecardpaymentgateway/classes/Response/FormPost.php  3
+ wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItem.php  6
+ wirecardpaymentgateway/classes/Response/Failure.php  3
+ wirecardpaymentgateway/helper/Service/OrderService.php  1
+ wirecardpaymentgateway/classes/Notification/Failure.php  4
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/NumericHelper.php  -1
- wirecardpaymentgateway/models/PaymentSepaDirectDebit.php  -2
- wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  -2
- wirecardpaymentgateway/classes/Notification/ProcessablePaymentNotificationFactory.php  -1
- wirecardpaymentgateway/helper/Form/FormHelper.php  -1
- wirecardpaymentgateway/classes/Response/ProcessablePaymentResponseFactory.php  -1
- wirecardpaymentgateway/models/PaymentCreditCard.php  -4
- wirecardpaymentgateway/helper/CurrencyHelper.php  -2
- wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItemCollection.php  -2
- wirecardpaymentgateway/models/PaymentSepaCreditTransfer.php  -2
- wirecardpaymentgateway/helper/CustomerHelper.php  -3
- wirecardpaymentgateway/models/PaymentIdeal.php  -2
- wirecardpaymentgateway/helper/OptionHelper.php  -1
- wirecardpaymentgateway/classes/Service/TransactionPossibleOperationService.php  -3
- wirecardpaymentgateway/helper/Form/Element/SwitchInput.php  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Coverage increased per file
===========================
+ wirecardpaymentgateway/models/PaymentGuaranteedInvoiceRatepay.php  3
+ wirecardpaymentgateway/classes/Response/Cancel.php  4
+ wirecardpaymentgateway/classes/Transaction/Builder/Entity/EntityBuilderFactory.php  19
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/classes/Response/Failure.php  3
+ wirecardpaymentgateway/classes/Response/Initial/Success.php  3
+ wirecardpaymentgateway/models/Payment.php  1
+ wirecardpaymentgateway/classes/Response/FormPost.php  3
+ wirecardpaymentgateway/helper/AdditionalInformationBuilder.php  4
+ wirecardpaymentgateway/classes/Notification/Success.php  4
+ wirecardpaymentgateway/classes/Config/CreditcardConfigurationFactory.php  9
+ wirecardpaymentgateway/helper/OrderManager.php  3
+ wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItem.php  6
+ wirecardpaymentgateway/models/CreditCardVault.php  2
+ wirecardpaymentgateway/classes/Service/TabManagerService.php  3
+ wirecardpaymentgateway/helper/Service/OrderService.php  1
+ wirecardpaymentgateway/helper/Service/ContextService.php  1
+ wirecardpaymentgateway/classes/Notification/Failure.php  4
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/models/PaymentCreditCard.php  -4
- wirecardpaymentgateway/helper/CurrencyHelper.php  -2
- wirecardpaymentgateway/classes/Response/ProcessablePaymentResponseFactory.php  -1
- wirecardpaymentgateway/helper/Form/FormHelper.php  -1
- wirecardpaymentgateway/helper/NumericHelper.php  -1
- wirecardpaymentgateway/models/PaymentSepaCreditTransfer.php  -2
- wirecardpaymentgateway/models/PaymentIdeal.php  -2
- wirecardpaymentgateway/helper/OptionHelper.php  -1
- wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  -2
- wirecardpaymentgateway/helper/Form/Element/SwitchInput.php  -4
- wirecardpaymentgateway/classes/Service/TransactionPossibleOperationService.php  -3
- wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItemCollection.php  -2
- wirecardpaymentgateway/models/PaymentSepaDirectDebit.php  -2
- wirecardpaymentgateway/helper/CustomerHelper.php  -3
- wirecardpaymentgateway/classes/Notification/ProcessablePaymentNotificationFactory.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Coverage increased per file
===========================
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItem.php  6
+ wirecardpaymentgateway/classes/Notification/Failure.php  4
+ wirecardpaymentgateway/models/PaymentGuaranteedInvoiceRatepay.php  3
+ wirecardpaymentgateway/classes/Notification/Success.php  4
+ wirecardpaymentgateway/helper/Service/ContextService.php  1
+ wirecardpaymentgateway/helper/Service/OrderService.php  1
+ wirecardpaymentgateway/classes/Response/Cancel.php  4
+ wirecardpaymentgateway/models/Payment.php  1
+ wirecardpaymentgateway/classes/Response/Initial/Success.php  3
+ wirecardpaymentgateway/classes/Response/Failure.php  3
+ wirecardpaymentgateway/classes/Service/TabManagerService.php  3
+ wirecardpaymentgateway/helper/OrderManager.php  3
+ wirecardpaymentgateway/classes/Transaction/Builder/Entity/EntityBuilderFactory.php  19
+ wirecardpaymentgateway/classes/Response/FormPost.php  3
+ wirecardpaymentgateway/helper/AdditionalInformationBuilder.php  4
+ wirecardpaymentgateway/models/CreditCardVault.php  2
+ wirecardpaymentgateway/classes/Config/CreditcardConfigurationFactory.php  9
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/models/PaymentSepaCreditTransfer.php  -2
- wirecardpaymentgateway/classes/Notification/ProcessablePaymentNotificationFactory.php  -1
- wirecardpaymentgateway/models/PaymentCreditCard.php  -4
- wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  -2
- wirecardpaymentgateway/helper/CustomerHelper.php  -3
- wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItemCollection.php  -2
- wirecardpaymentgateway/helper/Form/FormHelper.php  -1
- wirecardpaymentgateway/helper/NumericHelper.php  -1
- wirecardpaymentgateway/models/PaymentIdeal.php  -2
- wirecardpaymentgateway/classes/Service/TransactionPossibleOperationService.php  -3
- wirecardpaymentgateway/classes/Response/ProcessablePaymentResponseFactory.php  -1
- wirecardpaymentgateway/helper/OptionHelper.php  -1
- wirecardpaymentgateway/models/PaymentSepaDirectDebit.php  -2
- wirecardpaymentgateway/helper/Form/Element/SwitchInput.php  -4
- wirecardpaymentgateway/helper/CurrencyHelper.php  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/helper/OrderManager.php  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/helper/Service/ContextService.php  1
+ wirecardpaymentgateway/classes/Response/FormPost.php  3
+ wirecardpaymentgateway/classes/Notification/Failure.php  4
+ wirecardpaymentgateway/helper/AdditionalInformationBuilder.php  4
+ wirecardpaymentgateway/models/CreditCardVault.php  2
+ wirecardpaymentgateway/classes/Response/Failure.php  3
+ wirecardpaymentgateway/classes/Transaction/Builder/Entity/EntityBuilderFactory.php  19
+ wirecardpaymentgateway/models/Payment.php  1
+ wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItem.php  6
+ wirecardpaymentgateway/classes/Response/Initial/Success.php  3
+ wirecardpaymentgateway/classes/Config/CreditcardConfigurationFactory.php  9
+ wirecardpaymentgateway/helper/Service/OrderService.php  1
+ wirecardpaymentgateway/helper/OrderManager.php  6
+ wirecardpaymentgateway/classes/Response/Cancel.php  4
+ wirecardpaymentgateway/classes/Service/TabManagerService.php  3
+ wirecardpaymentgateway/classes/Notification/Success.php  4
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/models/PaymentGuaranteedInvoiceRatepay.php  3
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/Form/FormHelper.php  -1
- wirecardpaymentgateway/models/PaymentCreditCard.php  -4
- wirecardpaymentgateway/classes/Notification/ProcessablePaymentNotificationFactory.php  -1
- wirecardpaymentgateway/helper/CustomerHelper.php  -3
- wirecardpaymentgateway/models/PaymentIdeal.php  -2
- wirecardpaymentgateway/classes/Service/TransactionPossibleOperationService.php  -3
- wirecardpaymentgateway/helper/Form/Element/SwitchInput.php  -4
- wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  -2
- wirecardpaymentgateway/models/PaymentSepaDirectDebit.php  -2
- wirecardpaymentgateway/helper/OptionHelper.php  -1
- wirecardpaymentgateway/helper/CurrencyHelper.php  -2
- wirecardpaymentgateway/helper/NumericHelper.php  -1
- wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItemCollection.php  -2
- wirecardpaymentgateway/classes/Response/ProcessablePaymentResponseFactory.php  -1
- wirecardpaymentgateway/models/PaymentSepaCreditTransfer.php  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

File: wirecardpaymentgateway/wirecardpaymentgateway.php [hunk 10]
```diff
@@ -431,6 +429,17 @@ public function hookActionPaymentConfirmation($params)
         $this->displayName = $order->payment;
     }
 
+    /**
+     * Update “lang” tables after adding or updating a language
+     *
+     * @param $params
+     * @since 2.8.0
+     */
+    public function hookActionUpdateLangAfter($params)
```
Comment (rinnhofer) [inline wirecardpaymentgateway/wirecardpaymentgateway.php]: Codacy Issue found: [Avoid unused parameters such as '$params'.](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Coverage increased per file
===========================
+ wirecardpaymentgateway/helper/OrderManager.php  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/classes/Response/FormPost.php  3
+ wirecardpaymentgateway/helper/Service/OrderService.php  1
+ wirecardpaymentgateway/helper/AdditionalInformationBuilder.php  4
+ wirecardpaymentgateway/helper/OrderManager.php  7
+ wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItem.php  6
+ wirecardpaymentgateway/models/PaymentGuaranteedInvoiceRatepay.php  3
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/models/Payment.php  1
+ wirecardpaymentgateway/classes/Response/Initial/Success.php  3
+ wirecardpaymentgateway/classes/Response/Failure.php  3
+ wirecardpaymentgateway/classes/Service/TabManagerService.php  3
+ wirecardpaymentgateway/helper/Service/ContextService.php  1
+ wirecardpaymentgateway/classes/Notification/Failure.php  4
+ wirecardpaymentgateway/classes/Transaction/Builder/Entity/EntityBuilderFactory.php  19
+ wirecardpaymentgateway/classes/Response/Cancel.php  4
+ wirecardpaymentgateway/models/CreditCardVault.php  2
+ wirecardpaymentgateway/classes/Config/CreditcardConfigurationFactory.php  9
+ wirecardpaymentgateway/classes/Notification/Success.php  4
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/Form/Element/SwitchInput.php  -4
- wirecardpaymentgateway/models/PaymentSepaDirectDebit.php  -2
- wirecardpaymentgateway/classes/Service/TransactionPossibleOperationService.php  -3
- wirecardpaymentgateway/models/PaymentIdeal.php  -2
- wirecardpaymentgateway/models/PaymentSepaCreditTransfer.php  -2
- wirecardpaymentgateway/helper/OptionHelper.php  -1
- wirecardpaymentgateway/helper/NumericHelper.php  -1
- wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  -2
- wirecardpaymentgateway/helper/Form/FormHelper.php  -1
- wirecardpaymentgateway/helper/CustomerHelper.php  -3
- wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItemCollection.php  -2
- wirecardpaymentgateway/models/PaymentCreditCard.php  -4
- wirecardpaymentgateway/helper/CurrencyHelper.php  -2
- wirecardpaymentgateway/classes/Response/ProcessablePaymentResponseFactory.php  -1
- wirecardpaymentgateway/classes/Notification/ProcessablePaymentNotificationFactory.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/models/Payment.php  1
+ wirecardpaymentgateway/classes/Response/FormPost.php  3
+ wirecardpaymentgateway/helper/OrderManager.php  7
+ wirecardpaymentgateway/classes/Notification/Success.php  4
+ wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItem.php  6
+ wirecardpaymentgateway/models/CreditCardVault.php  2
+ wirecardpaymentgateway/classes/Service/TabManagerService.php  3
+ wirecardpaymentgateway/classes/Response/Cancel.php  4
+ wirecardpaymentgateway/classes/Transaction/Builder/Entity/EntityBuilderFactory.php  19
+ wirecardpaymentgateway/classes/Notification/Failure.php  4
+ wirecardpaymentgateway/models/PaymentGuaranteedInvoiceRatepay.php  3
+ wirecardpaymentgateway/classes/Response/Failure.php  3
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/helper/AdditionalInformationBuilder.php  4
+ wirecardpaymentgateway/helper/Service/ContextService.php  1
+ wirecardpaymentgateway/classes/Response/Initial/Success.php  3
+ wirecardpaymentgateway/helper/Service/OrderService.php  1
+ wirecardpaymentgateway/classes/Config/CreditcardConfigurationFactory.php  9
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/NumericHelper.php  -1
- wirecardpaymentgateway/helper/CustomerHelper.php  -3
- wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  -2
- wirecardpaymentgateway/helper/OptionHelper.php  -1
- wirecardpaymentgateway/classes/Transaction/Entity/Cart/CartItemCollection.php  -2
- wirecardpaymentgateway/helper/Form/FormHelper.php  -1
- wirecardpaymentgateway/classes/Service/TransactionPossibleOperationService.php  -3
- wirecardpaymentgateway/models/PaymentIdeal.php  -2
- wirecardpaymentgateway/models/PaymentSepaDirectDebit.php  -2
- wirecardpaymentgateway/classes/Notification/ProcessablePaymentNotificationFactory.php  -1
- wirecardpaymentgateway/classes/Response/ProcessablePaymentResponseFactory.php  -1
- wirecardpaymentgateway/helper/Form/Element/SwitchInput.php  -4
- wirecardpaymentgateway/helper/CurrencyHelper.php  -2
- wirecardpaymentgateway/models/PaymentSepaCreditTransfer.php  -2
- wirecardpaymentgateway/models/PaymentCreditCard.php  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 11]
```diff
@@ -23,10 +23,29 @@
...
+     * @var string
+     * @since 2.8.0
+     */
+    const TRANSLATION_FILE = "ordermanager";
+
     const WIRECARD_OS_STARTING = 'WIRECARD_OS_STARTING';
     const WIRECARD_OS_AWAITING = 'WIRECARD_OS_AWAITING';
     const WIRECARD_OS_AUTHORIZATION = 'WIRECARD_OS_AUTHORIZATION';
 
+    const WIRECARD_OS_TRANSLATION_KEY_STARTED = 'order_state_payment_started';
```
Comment (edra-bumci) [inline wirecardpaymentgateway/helper/OrderManager.php]: @sebastijansakac Are those constants used anywhere else apart from being mapped below in the ORDER_STATE_TRANSLATION_KEY_MAP? If not do we really need this extra step to assign those values to constants?

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 12]
```diff
@@ -71,65 +90,73 @@ public function createOrder($cart, $state, $paymentType)
...
+        foreach (\Language::getLanguages(false) as $language) {
+            $orderStateInfo = $this->module->getTranslationForLanguage(
+                $language['iso_code'],
+                $translationKey,
+                self::TRANSLATION_FILE
             );
+            $orderState->name[$language['id_lang']] = $orderStateInfo;
         }
+        $orderState->send_email = false;
+        $orderState->color = 'lightblue';
```
Comment (edra-bumci) [inline wirecardpaymentgateway/helper/OrderManager.php]: @sebastijansakac I know this was already there but can we maybe define it as a const?

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 13]
```diff
@@ -71,65 +90,73 @@ public function createOrder($cart, $state, $paymentType)
...
+        $orderState->hidden = false;
+        $orderState->delivery = false;
+        $orderState->logable = true;
+        $orderState->module_name = 'wirecardpaymentgateway';
```
Comment (edra-bumci) [inline wirecardpaymentgateway/helper/OrderManager.php]: @sebastijansakac I know this was already there but can we maybe define it as a const?

File: wirecardpaymentgateway/helper/OrderManager.php [hunk 14]
```diff
@@ -71,65 +90,73 @@ public function createOrder($cart, $state, $paymentType)
...
+
+    /**
+     * @param $state
+     * @return \OrderState
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     * @throws \Exception
+     * @since 2.8.0
+     */
+    private function createNewOrderState($state)
```
Comment (edra-bumci) [inline wirecardpaymentgateway/helper/OrderManager.php]: @sebastijansakac there are two functions which share a similar name and the purpose is not clear on why they differ. There is this function and the createOrderState. Can you maybe think of a way to rename them properly?

File: wirecardpaymentgateway/wirecardpaymentgateway.php [hunk 15]
```diff
@@ -1091,4 +1099,15 @@ private function initTabs()
...
         return $tabsConfig;
     }
+
+    /**
+     *  Add or update (if exists) order statuses
+     */
+    private function addUpdateOrderStatuses()
+    {
+        $orderManager = new OrderManager();
+        $orderManager->createOrderState(OrderManager::WIRECARD_OS_AUTHORIZATION);
```
Comment (edra-bumci) [inline wirecardpaymentgateway/wirecardpaymentgateway.php]: @sebastijansakac just an idea: would it be better to maybe call this function multiple times in a more generic way? Like for example use the array of key values:
        foreach (OrderManager::ORDER_STATE_TRANSLATION_KEY_MAP as $key=>$value)
	    {
		    $orderManager->createOrderState($key);
	    }

File: wirecardpaymentgateway/wirecardpaymentgateway.php [hunk 16]
```diff
@@ -1091,4 +1099,17 @@ private function initTabs()
...
     }
+
+    /**
+     *  Add or update (if exists) order statuses
+     */
+    private function addUpdateOrderStatuses()
+    {
+        $orderManager = new OrderManager();
+        $translationKeys = $orderManager::ORDER_STATE_TRANSLATION_KEY_MAP;
+        foreach ($translationKeys as $translationKey => $orderState)
```
Comment (rinnhofer) [inline wirecardpaymentgateway/wirecardpaymentgateway.php]: Codacy Issue found: [Expected 1 space(s) after closing parenthesis; found newline](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066&bid=16851682)

File: wirecardpaymentgateway/wirecardpaymentgateway.php (hunk 16 above)
Comment (rinnhofer) [inline wirecardpaymentgateway/wirecardpaymentgateway.php]: Codacy Issue found: [Avoid unused local variables such as '$orderState'.](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5127066)

**Pull request merged**